embassy-stm32/src/spi/mod.rs

```diff
@@ -476,6 +475,7 @@ impl<'d, T: Instance, Tx, Rx> Spi<'d, T, Tx, Rx> {
         let tx_f = crate::dma::write_repeated(&mut self.txdma, tx_request, &clock_byte, clock_byte_count, tx_dst);
 
         unsafe {
+            set_rxdmaen(T::REGS, true);
```

Comment:
there's some versions where the RM explicitly says you must set RXDMAEN=true, start DMAs, set TXDMAEN=true in that exact order. It was F4 iirc. And indeed if you don't do that it doesn't work.